test/integration/job/job_test.go

```diff
@@ -1182,11 +1182,17 @@ func setup(t *testing.T, nsBaseName string) (framework.CloseFunc, *restclient.Co
 	return closeFn, &config, clientSet, ns
 }
 
-func startJobController(restConfig *restclient.Config) (context.Context, context.CancelFunc) {
+func startJobControllerAndWaitForCaches(restConfig *restclient.Config) (context.Context, context.CancelFunc) {
 	informerSet := informers.NewSharedInformerFactory(clientset.NewForConfigOrDie(restclient.AddUserAgent(restConfig, "job-informers")), 0)
 	jc, ctx, cancel := createJobControllerWithSharedInformers(restConfig, informerSet)
 	informerSet.Start(ctx.Done())
 	go jc.Run(ctx, 1)
+
+	// since this method starts the controller in a separate goroutine
+	// and the tests don't check /readyz there is no way
+	// the tests can tell it is safe to call the server and requests won't be rejected
+	// thus we wait until caches have synced
+	informerSet.WaitForCacheSync(ctx.Done())
```

Comment:
Shouldn't this be done before starting the worker routines?

Reply:
the controller waits for the caches to be synced before starting the worker routines.

Reply:
So this is a bit of a hack for the tests to wait for the apiserver to be ready before sending requests... right?

Maybe the comment should have said:

```golang
// since the apiserver is started in a separate goroutine
```